pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3434,23 +3444,20 @@ public void endReplaceSegments(String tableNameWithType, String segmentLineageEn
 
         // Check that all the segments from 'segmentsTo' exist in the table
         Set<String> segmentsForTable = new HashSet<>(getSegmentsFor(tableNameWithType, false));
-        Preconditions.checkArgument(segmentsForTable.containsAll(lineageEntry.getSegmentsTo()), String.format(
-            "Not all segments from 'segmentsTo' are available in the table. (tableName = '%s', segmentsTo = '%s', "
-                + "segmentsFromTable = '%s')", tableNameWithType, lineageEntry.getSegmentsTo(), segmentsForTable));
+        List<String> segmentsTo = lineageEntry.getSegmentsTo();
```

Comment:
Do we want to make the same change for `updateSegmentLineageEntryToReverted()`?

Reply:
Good call, updated